### PR TITLE
Fixed - Infinite loading after deleting messages

### LINKED ISSUE #1982: Infinite loading after deleting messages
**Describe the bug**
Shows infinite loading after clicking the "Delete your messages" 

**To Reproduce**
1. Go to Messages
2. Message user
3. Click the menu icon dots
4. Click Delete your messages

**Expected behavior**
Will delete the messages 

**Screenshots**
https://prnt.sc/wdlxby

### PR DESCRIPTION
**Pull request by nitishkaila**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #1982.

### How to test the changes in this Pull Request:

1.Go to Messages
2. Message user
3. Click Delete your messages
4. Now compose screen should be displayed if no other thread found. or switch to the other thread if found.

### Proof Screenshots or Video
https://www.loom.com/share/097dfefbb76844938d90f77fddf81993

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-templates/bp-nouveau/includes/messages/ajax.php [hunk 1]
```diff
@@ -1211,9 +1211,10 @@ function bp_nouveau_ajax_delete_thread_messages() {
 
 	wp_send_json_success(
 		array(
-			'id'       => $thread_id,
-			'type'     => 'success',
-			'messages' => 'Messages successfully deleted.',
+			'id'       			=> $thread_id,
+			'type'     			=> 'success',
+			'messages'			=> 'Messages successfully deleted.',
+			'messages_count'	=> bp_get_message_thread_total_count( $thread_id ),
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/messages/ajax.php:1217]: @nitishkaila Run the PHPCBF to make tabbing proper aligned.

File: src/bp-templates/bp-nouveau/js/buddypress-messages.js [hunk 2]
```diff
@@ -2524,8 +2524,20 @@ window.bp = window.bp || {};
...
+							if ( undefined !== response.messages_count && 0 === response.messages_count ) {
+								if ( bp.Nouveau.Messages.threads.length > 1 ) {
+									bp.Nouveau.Messages.threads.remove( response.id );
+									bp.Nouveau.Messages.router.navigate( 'view/' + bp.Nouveau.Messages.threads.at( 0 ).id + '/', { trigger: true } );
+								} else {
+									BP_Nouveau.messages.hasThreads = false;
+									bp.Nouveau.Messages.threads.remove( response.id );
+									bp.Nouveau.Messages.router.navigate( 'view/?refresh=1', { trigger: true } );
+									bp.Nouveau.Messages.router.navigate( 'compose/', { trigger: true } );
+								}
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/js/buddypress-messages.js:2536]: @nitishkaila Run the PHPCBF to make tabbing proper aligned.

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
@nitishkaila Looks good to me.

Run the grunt on this branch and fix phpcs issues.

**Comment by nitishkaila** [comment]
@chetansatasiya 

I have run the Grunt and fixed the PHPCS issue.

File: src/bp-templates/bp-nouveau/js/buddypress-messages.js (hunk 2 above)
Comment (nitishkaila) [reply src/bp-templates/bp-nouveau/js/buddypress-messages.js:2536]: @chetansatasiya I have run the PHPCBF for this js file changes. But not found any changes.

**Review by KartikSuthar — APPROVED** [review]

**Comment by mohit-buddyboss** [comment]
@KartikSuthar 

FIXED> Tested this on dev branch of the platform. 
Previously, user was getting stuck in a loading mode if logged in user tries to deletes his/her messages using "Delete Your Messages"  option and if messages thread have messages from only one side (Example : IF A send messages to user B for the first time and then tries to delete messages using "Delete Your Messages" option) then on using "Delete your messages" message page was getting stuck in a loading mode.  
Now this issue resolve on dev branch. 

1. If message thread have 2 way conversation then on using "Delete your messages" option. Messages of logged in user will get delete successfully.
2. IF message thread have 1 way conversation( messages from logged in user only) then on using "Delete your messages" option. Messages thread will get delete and user will switch to first existing thread in the list. 
3. IF message thread have 1 way conversation( messages from logged in user only) then on using "Delete your messages" option. Messages thread will get delete and user will switch to compose messages screen. 
4. IF user is blocked by logged in user and thread have 1 way conversation( messages from logged in user only) then on using "Delete your messages" option. Messages thread will get delete and user will switch to first thread in the list or to compose message screen if user don't have any thread. 

Videos:
https://www.loom.com/share/64752113ce9c4de89bf448680f4a5b0f
https://www.loom.com/share/9970f95bb19a45e094cfac104217e35f

Related issue is resolve now and PR is ready to merge.

**Pull request merged**